src/Components/Bookings/Bookings.js

```diff
@@ -1,10 +1,178 @@
-import React from "react";
+import React, { useState, useContext, useEffect } from "react";
+import "./Bookings.css";
+import BookingsModal from "./BookingsModal/BookingsModal";
+import Backdrop from "./Backdrop/Backdrop";
+import makeRequest from ".././../Utils/index";
+import deleteBooking from "../.././Helpers/Delete";
+import editBooking from "../.././Helpers/Edit";
+import getCompletedTrips from "../../Helpers/completedTrip";
+import CompletedModal from "./CompletedTripModal/CompletedModal";
+import Authcontext from "../Context/context";
 
 const Bookings = (props) => {
+	const [modalViewing, setModalViewing] = useState(false);
+	const [bookings, setBookings] = useState([]);
+	const [selectedBooking, setSelectedBooking] = useState(null);
+	const [isEdit, setIsEdit] = useState(false);
+	const [isComplete, setIsComplete] = useState(null);
+	const [completedTrip, setCompletedTrip] = useState([]);
+	const [selectCompletedTrip, setSelectCompletedTrip] = useState(null);
+	const [completeView, setCompleteView] = useState(false);
+	const context = useContext(Authcontext);
+
+	useEffect(() => {
+		fetchBookings();
+		fetchCompleteTrips();
+	}, []);
+
+	const viewCompleteTrips = (bookingId) => {
+		setCompleteView(true);
+		const selected = completedTrip.find((e) => e.id === bookingId);
+		setSelectCompletedTrip(selected);
+	};
+	const closeCompletedModal = () => {
+		setCompleteView(false);
+	};
+
+	const modalCancelHandler = () => {
+		setModalViewing(false);
+	};
+	const onEditHandler = () => {
+		setIsEdit(true);
+		setModalViewing(false);
+	};
+	const onViewHandler = (bookingId) => {
+		setModalViewing(true);
+		const selected = bookings.find((e) => e.id === bookingId);
+		setSelectedBooking(selected);
+	};
+
+	const fetchBookings = async () => {
+		const requestBody = {
+			query: `
+			  query{
+					bookings{
+						id
+						user_id
+						trip
+						completed
+					}
+        }
+			`,
+		};
+		const result = await makeRequest({
+			data: requestBody,
+			token: context.token,
+		}).then((resData) => {
+			setBookings(resData.data.bookings);
+		});
+	};
+	const deleteHandler = async (bookingId) => {
```

Comment:
shouldn't be async since you're not "awaiting" **makeRequest**. In this case you chose the Native promise approach with **.then**

Reply:
resolved